session_0/item.rb

```diff
@@ -41,5 +47,9 @@ def discounted?
   # Note: If there are no items for category, stock price for category
   # should be zero.
   def self.stock_price_by_category(items)
+    stock_prices = {}
+    CATEGORIES.each {|key,value| stock_prices[key] = 0}
+    items.each {|item| stock_prices[item.category_id] += item.quantity * item.current_price}
```

Comment:
It looks slightly ugly because I am not used to seeing assignments within the `each` block. Maybe this looks better:

```ruby
items.each do |item|
  stock_prices[item.category_id] += item.quantity * item.current_price
end
```